node/store.go

```diff
@@ -56,12 +60,19 @@ func WithValueLogFileSize(size int64) StoreOpt {
 }
 
 // WithEncryptionKey sets the badger encryption key.
-func WithEncryptionKey(encryptionKey []byte) StoreOpt {
+func WithEncryptionKey(encryptionKey string) StoreOpt {
 	return func(o *StoreOptions) {
 		o.encryptionKey = encryptionKey
 	}
 }
 
+// WithIndexCacheSize sets the index cache size.
```

Comment:
suggestion:  This may be confused with thinking we have a cache for secondary indexes somewhere, suggest changing to  `WithIndexCacheSize sets the rootstore index cache size`